diff --git a/node_modules/immer/dist/types/types-external.d.ts b/node_modules/immer/dist/types/types-external.d.ts
index 446137e..1d53d3b 100644
--- a/node_modules/immer/dist/types/types-external.d.ts
+++ b/node_modules/immer/dist/types/types-external.d.ts
@@ -94,8 +94,8 @@ export interface IProduce {
 export interface IProduceWithPatches {
     <Recipe extends AnyFunc>(recipe: Recipe): InferCurriedFromRecipe<Recipe, true>;
     <State, Recipe extends Function>(recipe: Recipe, initialState: State): InferCurriedFromInitialStateAndRecipe<State, Recipe, true>;
-    <Base, D = Draft<Base>>(base: Base, recipe: (draft: D) => ValidRecipeReturnType<Base>, listener?: PatchListener): PatchesTuple<Base>;
-    <Base, D = Draft<Base>>(base: Base, recipe: (draft: D) => Promise<ValidRecipeReturnType<Base>>, listener?: PatchListener): PatchesTuple<Promise<Base>>;
+    <Base, D = Draft<Base>>(base: Base, recipe: (draft: D) => ValidRecipeReturnType<D>, listener?: PatchListener): PatchesTuple<Base>;
+    <Base, D = Draft<Base>>(base: Base, recipe: (draft: D) => Promise<ValidRecipeReturnType<D>>, listener?: PatchListener): Promise<PatchesTuple<Base>>;
 }
 export declare function never_used(): void;
 export {};
